### PR TITLE
EIT-2497: Documentation - Cash App Pay

### PR DESCRIPTION
**Pull request by ScottAntonacAP**
## Summary of Changes

- Added documentation for Cash App Pay
- Updated GitHub pages dependencies

## Items of Note

The majority of these docs are ported from the Cash App PayKit for Android documentation, with required changes made for using with an Afterpay merchant account. To run the pages locally, change into the `/docs` directory and run `bundle exec jekyll serve`. Alternatively, to view them download and extract the zip containing the generated html.

[cash-app-pay-docs.zip](https://github.com/afterpay/sdk-android/files/10820301/cash-app-pay-docs.zip)

## Submission Checklist

- [x] Documentation is changed or added.

### CODE REVIEW

File: docs/getting-started/cash-app-pay.md [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+or with the async version of it:
+
+``` kotlin
+Afterpay.signCashAppOrderAsync(token)
+```
+
+### Step 5C: Create a PayKit Customer Request
+
+To charge $5.00 as a one-time payment, your createrequest call might look like this (in the below example, `cashAppData` is the response object that is returned in the `didReceiveCashAppData` handler method):
```
Comment (benfterpay) [inline docs/getting-started/cash-app-pay.md:162]: Does the amount need to be set in the server-to-server call? From from glancing through this, mentioning the amount made me expect to see it in the code. What about this?
```suggestion
To charge a one-time payment, your create request call might look like this (in the below example, `cashAppData` is the response object that is returned in the `didReceiveCashAppData` handler method, which would echo the amount from your server-to-server Create Checkout call):
```

File: docs/getting-started/cash-app-pay.md [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+### Step 6A: Add an Authorize Request Event to Cash App Pay button
+
+Once the Cash App PayKit SDK is in the `ReadyToAuthorize` state, you can display the Cash App Pay button. When the Customer taps the button, you can authorize the customer request. Documentation for a Cash App Pay button component can be found [here][cash-button-docs]{:target='_blank'}
+
+``` kotlin
+payKit.authorizeCustomerRequest()
+```
+
+{: .info }
+> Currently, the Button provided by the SDK is unmanaged. This means that it's a stylized button which isn't aware of SDK events out-of-the-box. It is the developer's responsibility to call the above method when the button is clicked and also manage any disabled and loading states.
```
Comment (benfterpay) [inline docs/getting-started/cash-app-pay.md:186]:
```suggestion
> Currently, the Button provided by the SDK is unmanaged. This means that it's a stylized button which isn't aware of SDK events out-of-the-box. It is the developer's responsibility to call the above method when the button is tapped and also manage any disabled and loading states.
```

File: docs/getting-started/cash-app-pay.md [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    when (validationResult) {
+      is CashAppValidationResponse.Success -> TODO("Capture payment with token and grant id")
+      is CashAppValidationResponse.Failure -> TODO("Handle an invalid Cash App order")
+    }
+}
+```
+
+## Step 7: Pass Grants to the Backend and Capture Payment
+
+The `Approved` state will contain a Grants list object associated with it and it can be used with Afterpay's capture payment API. Pass the grant id along with the token to capture using a server to server request.
```
Comment (benfterpay) [inline docs/getting-started/cash-app-pay.md:212]:
```suggestion
The `Approved` state will contain a Grants list object associated with it and it can be used with Afterpay's "Immediate Payment Capture" or "Deferred Payment Auth" API. Pass the grant ID along with the token to capture/auth using a server-to-server request.
```

**Review by benfterpay — CHANGES_REQUESTED** [review]

File: docs/getting-started/cash-app-pay.md [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+val payKit : CashAppPayKit = CashAppPayKitFactory.createSandbox(Afterpay.environment.payKitId)
+```
+
+Creating a production Cash App PayKit SDK instance:
+``` kotlin
+val payKit : CashAppPayKit = CashAppPayKitFactory.create(Afterpay.environment.payKitId)
+```
+
+{: .info }
+**Optional:** We recommend using the [Cash App Sandbox App][sandbox-app]{:target="_blank"} to compare the Sandbox environment of the SDK during your development phase.
```
Comment (benfterpay) [inline docs/getting-started/cash-app-pay.md:70]:
```suggestion
**Optional:** We recommend using the [Cash App Sandbox App][sandbox-app]{:target="_blank"} to test the payment flow in the Sandbox environment during your development phase.
```

File: docs/getting-started/cash-app-pay.md [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+```
+
+For definitions of other build systems, see [Cash PayKit on Maven Central][cash-on-maven]{:target="_blank"}.
+
+{: .info }
+> Version `v1.0.3` of the SDK size is `12.3 kB`.
+
+## Step 2: Create a Cash App PayKit SDK Instance
+
+To create a new instance of the Cash App PayKit SDK, you must pass the `clientId`. This is a required field. This can be retrieved through the Afterpay object: `Afterpay.environment.payKitId`.
```
Comment (pedronveloso) [inline docs/getting-started/cash-app-pay.md:47]: I wonder if it would make sense for us to rename `payKitId` to `payKitClientId`, for having a better 1-1 correspondence. WDYT?

File: docs/getting-started/cash-app-pay.md [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+
+{: .note }
+> Ensure that the Afterpay SDK is configured per the [instructions][configure-afterpay] before attempting to access `Afterpay.environment.payKitId`
+
+
+You should use CashAppPayKitFactory to create an instance of the Cash App PayKit SDK. When doing so, you'll need to specify the environment you will use, Sandbox or Production. The function `createSandbox()` will create an SDK instance in the Sandbox environment.
```
Comment (pedronveloso) [inline docs/getting-started/cash-app-pay.md:53]: nit: add backticks to `CashAppPayKitFactory`

File: docs/getting-started/cash-app-pay.md [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+  {: .text-delta }
+- TOC
+{:toc}
+</details>
+
+
+{: .alert }
+> Cash App Pay is currently available in the following region(s): US
+
+Cash App Pay support can be added using your Afterpay merchant account. This requires a token to be generated by a server-to-server call to the Afterpay API Create Checkout endpoint with a parameter of `isCashAppPay` set to `true`. This method requires importing and implementing the Cash App PayKit SDK.
```
Comment (emgroves) [inline docs/getting-started/cash-app-pay.md:27]: nit: `PayKit` should be `Pay Kit` where used

File: docs/getting-started/cash-app-pay.md [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+payKit.unregisterFromStateUpdates()
+```
+
+### States
+
+`PayKitState` is a sealed class parameter. We suggest that you use a Kotlin `when` statement on it. Some of these possible states are for information only, but most drive the logic of your integration. The most critical states to handle are in the table below:
+
+| State  | Description |
+|:-------|:------------|
+| `ReadyToAuthorize` | Show a Cash App Pay button in your UI and call `authorizeCustomerRequest()` when it is tapped. |
```
Comment (pedronveloso) [inline docs/getting-started/cash-app-pay.md:100]: I guess I need to correct this on our docs as well. The sentence here is a bit misleading, since it **sounds** like this state will show the button, but in reality what we meant to say here is that it is up to the developer to show the button at this point. So perhaps let's rephrase this as: "You should show the Cash App Pay button ...." .

I'll do the same update on our side.

**Review by pedronveloso — APPROVED** [review]
LGTM. Made a few comments for possible improvements, but they're minor things.

File: docs/getting-started/cash-app-pay.md (hunk 7 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: Addressed in [bf06244](https://github.com/afterpay/sdk-android/pull/185/commits/bf062444eb308c1ffa171aa93951333bc266c7cb)

File: docs/getting-started/cash-app-pay.md (hunk 5 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: Changed for docs in [a0b7a0f](https://github.com/afterpay/sdk-android/pull/185/commits/a0b7a0f60c793dc140b969b286d60c7951eb721d). Will update code in #184

File: docs/getting-started/cash-app-pay.md (hunk 6 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: Fixed in [9c0ab09](https://github.com/afterpay/sdk-android/pull/185/commits/9c0ab09064b7fc70ced2fbcf05a8f0c94112c7cc)

**Review by benfterpay — APPROVED** [review]

File: docs/getting-started/cash-app-pay.md [hunk 9]
```diff
@@ -24,46 +24,46 @@ NEW (v4.3.0)
 {: .alert }
 > Cash App Pay is currently available in the following region(s): US
 
-Cash App Pay support can be added using your Afterpay merchant account. This requires a token to be generated by a server-to-server call to the Afterpay API Create Checkout endpoint with a parameter of `isCashAppPay` set to `true`. This method requires importing and implementing the Cash App PayKit SDK.
+Cash App Pay support can be added using your Afterpay merchant account. This requires a token to be generated by a server-to-server call to the Afterpay API Create Checkout endpoint with a parameter of `isCashAppPay` set to `true`. This method requires importing and implementing the Cash App Pay Kit SDK.
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:27]:
```suggestion
You can addd Cash App Pay support using your Afterpay merchant account. To do this, you must generate a token by sending a server-to-server call to the Afterpay API Create Checkout <!--can you provide a link to this here?--> endpoint with the parameter `isCashAppPay` set to `true`. This method requires importing and implementing the Cash App Pay Kit SDK.
```

File: docs/getting-started/cash-app-pay.md [hunk 10]
```diff
@@ -24,46 +24,46 @@ NEW (v4.3.0)
...
 
-## Step 2: Create a Cash App PayKit SDK Instance
+## Step 2: Create a Cash App Pay Kit SDK Instance
 
-To create a new instance of the Cash App PayKit SDK, you must pass the `clientId`. This is a required field. This can be retrieved through the Afterpay object: `Afterpay.environment.payKitId`.
+To create a new instance of the Cash App Pay Kit SDK, you must pass the `clientId`. This is a required field. This can be retrieved through the Afterpay object: `Afterpay.environment.payKitClientId`.
 
 {: .note }
-> Ensure that the Afterpay SDK is configured per the [instructions][configure-afterpay] before attempting to access `Afterpay.environment.payKitId`
+> Ensure that the Afterpay SDK is configured per the [instructions][configure-afterpay] before attempting to access `Afterpay.environment.payKitClientId`
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:50]:
```suggestion
> Confirm that the Afterpay SDK is configured per the [instructions][configure-afterpay] before attempting to access `Afterpay.environment.payKitClientId`
```

File: docs/getting-started/cash-app-pay.md [hunk 11]
```diff
@@ -24,46 +24,46 @@ NEW (v4.3.0)
 {: .alert }
 > Cash App Pay is currently available in the following region(s): US
 
-Cash App Pay support can be added using your Afterpay merchant account. This requires a token to be generated by a server-to-server call to the Afterpay API Create Checkout endpoint with a parameter of `isCashAppPay` set to `true`. This method requires importing and implementing the Cash App PayKit SDK.
+Cash App Pay support can be added using your Afterpay merchant account. This requires a token to be generated by a server-to-server call to the Afterpay API Create Checkout endpoint with a parameter of `isCashAppPay` set to `true`. This method requires importing and implementing the Cash App Pay Kit SDK.
 
 {: .info }
 > When creating a checkout token, both `redirectConfirmUrl` and `redirectCancelUrl` must be set. Failing to do so will cause undefined behavior. The SDK’s example merchant server sets the parameters [here][example-server-props]{:target='_blank'}. See more by checking the [api reference][api-reference-props]{:target='_blank'}.
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:30]:
```suggestion
> When creating a checkout token, you must set both `redirectConfirmUrl` and `redirectCancelUrl`. If they are not set, it will cause undefined behavior<!--can you be more specific on "undefined behavior"-->. The SDK’s example merchant server sets the parameters [here][example-server-props]{:target='_blank'}. See more details at [api reference][api-reference-props]{:target='_blank'}.
```

File: docs/getting-started/cash-app-pay.md [hunk 12]
```diff
@@ -24,46 +24,46 @@ NEW (v4.3.0)
...
+To create a new instance of the Cash App Pay Kit SDK, you must pass the `clientId`. This is a required field. This can be retrieved through the Afterpay object: `Afterpay.environment.payKitClientId`.
 
 {: .note }
-> Ensure that the Afterpay SDK is configured per the [instructions][configure-afterpay] before attempting to access `Afterpay.environment.payKitId`
+> Ensure that the Afterpay SDK is configured per the [instructions][configure-afterpay] before attempting to access `Afterpay.environment.payKitClientId`
 
 
-You should use CashAppPayKitFactory to create an instance of the Cash App PayKit SDK. When doing so, you'll need to specify the environment you will use, Sandbox or Production. The function `createSandbox()` will create an SDK instance in the Sandbox environment.
+You should use `CashAppPayKitFactory` to create an instance of the Cash App Pay Kit SDK. When doing so, you'll need to specify the environment you will use, Sandbox or Production. The function `createSandbox()` will create an SDK instance in the Sandbox environment.
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:53]:
```suggestion
You should use `CashAppPayKitFactory` to create an instance of the Cash App Pay Kit SDK. When doing so, you must specify the environment you will use, Sandbox or Production. The function `createSandbox()` will create an SDK instance in the Sandbox environment.
```

File: docs/getting-started/cash-app-pay.md [hunk 13]
```diff
@@ -24,46 +24,46 @@ NEW (v4.3.0)
...
 
 {: .info }
 > You should use the Sandbox environment during the development phase and Production for your actual production releases.
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:56]:
```suggestion
> You should use the Sandbox environment during the development phase and the Production environment for your production releases.
```

File: docs/getting-started/cash-app-pay.md [hunk 14]
```diff
@@ -24,46 +24,46 @@ NEW (v4.3.0)
...
 
-Creating a sandbox Cash App PayKit SDK instance:
+Creating a sandbox Cash App Pay Kit SDK instance:
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:58]:
```suggestion
Creating a Sandbox Cash App Pay Kit SDK instance:
```

File: docs/getting-started/cash-app-pay.md [hunk 15]
```diff
@@ -24,46 +24,46 @@ NEW (v4.3.0)
...
 
 ``` kotlin
-val payKit : CashAppPayKit = CashAppPayKitFactory.createSandbox(Afterpay.environment.payKitId)
+val payKit : CashAppPayKit = CashAppPayKitFactory.createSandbox(Afterpay.environment.payKitClientId)
 ```
 
-Creating a production Cash App PayKit SDK instance:
+Creating a production Cash App Pay Kit SDK instance:
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:64]:
```suggestion
Creating a Production Cash App Pay Kit SDK instance:
```

File: docs/getting-started/cash-app-pay.md [hunk 16]
```diff
@@ -97,7 +97,7 @@ payKit.unregisterFromStateUpdates()
 
 | State  | Description |
 |:-------|:------------|
-| `ReadyToAuthorize` | Show a Cash App Pay button in your UI and call `authorizeCustomerRequest()` when it is tapped. |
+| `ReadyToAuthorize` | You should show the Cash App Pay button in your UI and call `authorizeCustomerRequest()` when it is tapped. |
 | `Approved` | Grants are ready for your backend to use to create a payment. |
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:101]:
```suggestion
| `Approved` | Grants are ready for your backend to use and to create a payment. |
```

File: docs/getting-started/cash-app-pay.md [hunk 17]
```diff
@@ -126,40 +126,33 @@ Here’s an example of how this integration looks for your `AndroidManifest`:
...
-      }
-    }
-  })
-}
-```
-
-### Step 5B: Sign the Order Token
+### Step 5A: Sign the Order Token
 
 After retrieving the token from your server-to-server call, the order needs to be signed, so as to retrieve the JWT and associated data. This can be done either by the suspending function:
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:131]:
```suggestion
After retrieving the token from your server-to-server call, you must sign the order, so that you can retrieve the JWT and associated data. This can be done either by the suspending function or with the async version of it. 
```

File: docs/getting-started/cash-app-pay.md [hunk 18]
```diff
@@ -126,40 +126,33 @@ Here’s an example of how this integration looks for your `AndroidManifest`:
...
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:132]: Example of the suspending function:

File: docs/getting-started/cash-app-pay.md [hunk 19]
```diff
@@ -126,40 +126,33 @@ Here’s an example of how this integration looks for your `AndroidManifest`:
...
-Afterpay.signCashAppOrder(token)
+Afterpay.signCashAppOrder(token) { cashAppData ->
+  when (cashAppData) {
+    is CashAppSignOrderResult.Success -> TODO("Create the Pay Kit customemr request")
+    is CashAppSignOrderResult.Failure -> TODO("Display an error and restart payment flow")
+  }
+}
 ```
 
 or with the async version of it:
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:142]:
```suggestion
Example of the asynchronous version of the suspending function:
```

File: docs/getting-started/cash-app-pay.md [hunk 20]
```diff
@@ -126,40 +126,33 @@ Here’s an example of how this integration looks for your `AndroidManifest`:
...
+    is CashAppSignOrderResult.Failure -> TODO("Display an error and restart payment flow")
+  }
+}
 ```
 
-### Step 5C: Create a PayKit Customer Request
+### Step 5B: Create a Pay Kit Customer Request
 
-To charge a one-time payment, your create request call might look like this (in the below example, `cashAppData` is the response object that is returned in the `didReceiveCashAppData` handler method, which would echo the amount from your server-to-server Create Checkout call):
+To charge a one-time payment, your create request call might look like this (in the below example, `cashAppData` is the response object that is returned in the traling lambda in step 5A, which would echo the amount from your server-to-server Create Checkout call):
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:155]:
```suggestion
To charge a one-time payment, your `create request` call might look like this (in the following example, `cashAppData` is the response object that is returned in the trailing lambda in step 5A, which would echo the amount from your server-to-server `Create Checkout` call): <!--Please verify capitalization of `create checkout`-->
```

File: docs/getting-started/cash-app-pay.md [hunk 21]
```diff
@@ -176,7 +169,7 @@ payKit.createCustomerRequest(request)
 
 ### Step 6A: Add an Authorize Request Event to Cash App Pay button
 
-Once the Cash App PayKit SDK is in the `ReadyToAuthorize` state, you can display the Cash App Pay button. When the Customer taps the button, you can authorize the customer request. Documentation for a Cash App Pay button component can be found [here][cash-button-docs]{:target='_blank'}
+Once the Cash App Pay Kit SDK is in the `ReadyToAuthorize` state, you can display the Cash App Pay button. When the Customer taps the button, you can authorize the customer request. Documentation for a Cash App Pay button component can be found [here][cash-button-docs]{:target='_blank'}
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:172]:
```suggestion
Once the Cash App Pay Kit SDK is in the `ReadyToAuthorize` state, you can display the Cash App Pay button. When the customer taps the button, you can authorize the customer request. See [Cash Button Docs][cash-button-docs]{:target='_blank'} to learn more about the Cash App Pay button component.
```

**Review by jay-saldanha — COMMENTED** [review]
AWC

File: docs/getting-started/cash-app-pay.md (hunk 20 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: Addressed in [3f0282c](https://github.com/afterpay/sdk-android/pull/185/commits/3f0282c492fe293fd098fd2f2b4b0912de2acf1e)

Create Checkout is the name of a call, not code. Therefore it shouldn't be wrapped in backticks but will upper the Create Request case to match Create Checkout. The call to those functions will be implementation dependent and therefore not able to give a concrete piece of code out of concern for causing confusion.

File: docs/getting-started/cash-app-pay.md (hunk 18 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md:134]: Addressed in [3f0282c](https://github.com/afterpay/sdk-android/pull/185/commits/3f0282c492fe293fd098fd2f2b4b0912de2acf1e)

File: docs/getting-started/cash-app-pay.md (hunk 9 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: Addressed in [3f0282c](https://github.com/afterpay/sdk-android/pull/185/commits/3f0282c492fe293fd098fd2f2b4b0912de2acf1e)

File: docs/getting-started/cash-app-pay.md (hunk 11 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: Addressed in [3f0282c](https://github.com/afterpay/sdk-android/pull/185/commits/3f0282c492fe293fd098fd2f2b4b0912de2acf1e)

File: docs/getting-started/cash-app-pay.md (hunk 20 above)
Comment (jay-saldanha) [reply docs/getting-started/cash-app-pay.md]: ok

File: docs/getting-started/cash-app-pay.md (hunk 11 above)
Comment (jay-saldanha) [reply docs/getting-started/cash-app-pay.md]: @ScottAntonacAP Can you change the last line to See more details at [Redirect Method]([api-reference-props]{:target='_blank'} in the Standard Checkout API.

File: docs/getting-started/cash-app-pay.md (hunk 9 above)
Comment (jay-saldanha) [reply docs/getting-started/cash-app-pay.md]: In the first line, addd has three d's please remove one. sorry for missing this

File: docs/getting-started/cash-app-pay.md [hunk 22]
```diff
@@ -0,0 +1,213 @@
...
+
+{: .info }
+> Version `v1.0.3` of the SDK size is `12.3 kB`.
+
+## Step 2: Create a Cash App Pay Kit SDK Instance
+
+To create a new instance of the Cash App Pay Kit SDK, you must pass the `clientId`. This is a required field. This can be retrieved through the Afterpay object: `Afterpay.environment.payKitClientId`.
+
+{: .note }
+> Ensure that the Afterpay SDK is configured per the [instructions][configure-afterpay] before attempting to access `Afterpay.environment.payKitClientId`
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md]: @ScottAntonacAP the link to instructions is broken, please check

File: docs/getting-started/cash-app-pay.md (hunk 9 above)
Comment (jay-saldanha) [reply docs/getting-started/cash-app-pay.md]: @ScottAntonacAP I think the first line needs to be revised, my understanding is we are wrapping CA SDK inside the AP SDK, when we say "support" it is confusing.

File: docs/getting-started/cash-app-pay.md [hunk 23]
```diff
@@ -0,0 +1,213 @@
...
+}
+```
+
+You register with the SDK instance you’ve created above:
+
+``` kotlin
+payKit.registerForStateUpdates(this)
+```
+
+You should also use the unregister function when you're done with the SDK:
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:88]:
```suggestion
You should also use the `unregister` function when you're done with the SDK:
```

File: docs/getting-started/cash-app-pay.md [hunk 24]
```diff
@@ -0,0 +1,213 @@
...
+`PayKitState` is a sealed class parameter. We suggest that you use a Kotlin `when` statement on it. Some of these possible states are for information only, but most drive the logic of your integration. The most critical states to handle are in the table below:
+
+| State  | Description |
+|:-------|:------------|
+| `ReadyToAuthorize` | You should show the Cash App Pay button in your UI and call `authorizeCustomerRequest()` when it is tapped. |
+| `Approved` | Grants are ready for your backend to use to create a payment. |
+| `Declined` | Customer has declined the Cash App Pay authorization and must start the flow over or choose a new payment method. |
+| `PayKitExceptionState` | The general wrapper state for exceptions. These can range from integration errors to network errors. The exception states are emitted only for unrecoverable error states. |
+
+## Step 4: Implement Deep Linking Handling
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:105]:
```suggestion
## Step 4: Implement Deep Linking
```

File: docs/getting-started/cash-app-pay.md (hunk 21 above)
Comment (jay-saldanha) [reply docs/getting-started/cash-app-pay.md]: Please change the last line to " See [[CashPayKitButton][cash-button-docs]{:target='_blank'} to learn more about the Cash App Pay button component."

File: docs/getting-started/cash-app-pay.md [hunk 25]
```diff
@@ -0,0 +1,213 @@
...
+> Currently, the Button provided by the SDK is unmanaged. This means that it's a stylized button which isn't aware of SDK events out-of-the-box. It is the developer's responsibility to call the above method when the button is tapped and also manage any disabled and loading states.
+
+Your app will redirect to Cash App for authorization. When the authorization is completed, your redirect URI will be called to open your app. The SDK will fetch your authorized request and return it to your callback listener as one of 2 states: `Approved` or `Declined`.
+
+### Step 6B: Validate the Cash App Pay Order
+
+{: .alert }
+> This step must not be skipped
+
+Finally the Cash App order must be validated. This looks something like the below:
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:190]:
```suggestion
Finally, you must validate the Cash App order. This will look like the following example:
```

File: docs/getting-started/cash-app-pay.md [hunk 26]
```diff
@@ -0,0 +1,213 @@
...
+    when (validationResult) {
+      is CashAppValidationResponse.Success -> TODO("Capture payment with token and grant id")
+      is CashAppValidationResponse.Failure -> TODO("Handle an invalid Cash App order")
+    }
+}
+```
+
+## Step 7: Pass Grants to the Backend and Capture Payment
+
+The `Approved` state will contain a Grants list object associated with it and it can be used with Afterpay's "Immediate Payment Capture" or "Deferred Payment Auth" API. Pass the grant ID along with the token to capture/auth using a server-to-server request.
```
Comment (jay-saldanha) [inline docs/getting-started/cash-app-pay.md:207]:
```suggestion
The `Approved` state will contain a `Grants list` object associated with it and it can be used with Afterpay's **Immediate Payment Capture** or **Deferred Payment Auth** API. Pass the `grant ID` along with the token to capture/authorize using a server-to-server request.
```

**Review by jay-saldanha — COMMENTED** [review]
@ScottAntonacAP Added some more comments after reviewing the HTML file

File: docs/getting-started/cash-app-pay.md (hunk 9 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: We are not wrapping the Cash App Pay Kit SDK, but rather adding allowing the merchants app to support Cash App Pay using their Afterpay merchant account. That's why it needs to be imported. Happy to update the first line 🙂 but given that information, what would you suggest?

File: docs/getting-started/cash-app-pay.md (hunk 22 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: Thanks, the link is only broken in the html that I passed across because it was a saved file that didn't include the whole site. The link works on my local with the dev server running and will work once live. Really appreciate the thorough testing 👍

File: docs/getting-started/cash-app-pay.md (hunk 23 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: Addressed in https://github.com/afterpay/sdk-android/pull/185/commits/522aa7b27c98d5b78b5cc00a438de93adecf5e74

Again, register is not a function, so updated to bold text and capitalised.

File: docs/getting-started/cash-app-pay.md (hunk 26 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: Addressed in [b69e03e](https://github.com/afterpay/sdk-android/pull/185/commits/b69e03ec5a358005cc5e5823c75b8114e7b58f05)

File: docs/getting-started/cash-app-pay.md (hunk 11 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: Addressed in [522aa7b](https://github.com/afterpay/sdk-android/pull/185/commits/522aa7b27c98d5b78b5cc00a438de93adecf5e74)

File: docs/getting-started/cash-app-pay.md (hunk 9 above)
Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: Potentially the first line could just be changed to something like:

> You can add Cash App Pay support to your app using your Afterpay merchant account.

Comment (jay-saldanha) [reply docs/getting-started/cash-app-pay.md]: With our latest enhancements, you can now support taking Cash App Pay payments using your Afterpay merchant account.  - does this sound better?

Comment (jay-saldanha) [reply docs/getting-started/cash-app-pay.md]: You can support taking Cash App Pay payments using your Afterpay merchant account. or With our latest enhancements, you can now support taking Cash App Pay payments using your Afterpay merchant account.

Comment (jay-saldanha) [reply docs/getting-started/cash-app-pay.md]: choose any one

Comment (ScottAntonacAP) [reply docs/getting-started/cash-app-pay.md]: Addressed in [b469f13](https://github.com/afterpay/sdk-android/pull/185/commits/b469f1364f0069df608df43ababea3e33ac7bf5b)

**Review by jay-saldanha — APPROVED** [review]
